azure-iot-device/azure/iot/device/common/auth/sastoken.py

```diff
@@ -94,3 +94,49 @@ def _build_token(self):
     def expiry_time(self):
         """Expiry Time is READ ONLY"""
         return self._expiry_time
+
+
+class NonRenewableSasToken(object):
+    def __init__(self, sastoken_string):
+        self._token = sastoken_string
+        self._token_info = get_sastoken_info_from_string(self._token)
+
+    def __str__(self):
+        return self._token
+
+    @property
+    def expiry_time(self):
+        """Expiry Time is READ ONLY"""
+        return int(self._token_info["se"])
+
+    @property
+    def resource_uri(self):
+        "Resource URI is READ ONLY"
```

Comment:
>"Resource URI is READ ONLY" [](start = 8, length = 27)

one quote mark or 3?  I think either woks most of time, but I think 3 is convention for docstrings so some tools might not identify this as a doc string.  Or maybe not.

Reply:
Good catch - it works either way, but yeah you run into problems when you leave out the convention like that.

---
In reply to: [506677009](https://github.com/Azure/azure-iot-sdk-python/pull/692#discussion_r506677009) [](ancestors = 506677009)